### DISCUSSION TITLE
React NavigationのRouteを学習コンテンツの応用編に入れておくか？

### DISCUSSION THREAD

**Discussion by kozake** [ws-4020/mobile-app-crib-notes#293] [General]
応用編の「遷移先の画面にパラメータを渡す」について、TODOアプリの実装として使う場面はなかったですが、基本的なアプリ開発でよく使う機能ではないでしょうか。
「TODOアプリの実装では使わないですが」の但し書きをしたうえで、「React Navigationの基本」に含めてもいいように感じました。

**Comment by tani-takanori** [comment]
基本に入れるとすると...

- Overview
- Stack
- Modal
- Tab
- Route

みたいになるのかな？

**Comment by tani-takanori** [comment]
> 応用編の「遷移先の画面にパラメータを渡す」について、TODOアプリの実装として使う場面はなかったですが、基本的なアプリ開発でよく使う機能ではないでしょうか。

この指摘に対応するとなると「基本的なアプリ開発でよく使う機能」を入れておかないとあかんのかな？

**Comment by tani-takanori** [comment]
Routeを使う場面やシチュエーションが整理できないと「基本なのか？」がわからない。

「React Navigationの基本」に入れるなら、そもそも学習コンテンツから落としていいのでは？

Reduxが市民権を得ている中でRouteを使うのか？

**Comment by tani-takanori** [comment]
画面毎のパラメータをどうやって管理するかをSPAと対比して考えたほうがいいんじゃないか？

**Comment by tani-takanori** [comment]
# 応用編にRouteを追加した経緯

Routeを入れていたのはReact Navigationの型定義を省略したため、画面の型（画面名＋パラメータ）定義がなくなってしまった。
そこの学習を補完するために応用編を追加した

**Reply by tani-takanori** [reply]
型定義の複雑さを基本に入れてしまうことになるので、応用編に入れておいたらいいのではないか？

**Comment by creature-water-valley** [comment]
Q&Aアプリの知識として必要だったため、アプリの実装前に学習するコンテンツの中に移動
- https://github.com/ws-4020/mobile-app-crib-notes/pull/1255
